src/main/io/gps.c

```diff
@@ -1805,22 +2352,19 @@ static bool gpsNewFrameUBLOX(uint8_t data)
             _step++;
             _ck_b += (_ck_a += data);       // checksum byte
             _payload_length += (uint16_t)(data << 8);
-#if DEBUG_UBLOX_FRAMES
-    debug[3] = _payload_length;
-#endif
-            if (_payload_length > UBLOX_PAYLOAD_SIZE) {
-                _skip_packet = true;
-            }
+
             _payload_counter = 0;   // prepare to receive payload
             if (_payload_length == 0) {
                 _step = 7;
             }
+            _buffer.bytes[_payload_counter] = data;
+            if (_payload_length > UBLOX_PAYLOAD_SIZE) {
+                _payload_length = UBLOX_PAYLOAD_SIZE;
+            }
             break;
         case 6:
             _ck_b += (_ck_a += data);       // checksum byte
-            if (_payload_counter < UBLOX_PAYLOAD_SIZE) {
```

Comment:
Checksum all packet bytes, but store only the ones that fit into _buffer

Reply:
Yes that is what is needed.

Reply:
> Checksum all packet bytes, but store only the ones that fit into _buffer

That's what was happening with the "adam-ah suggestion for payload optimisation" commit.
The "adam-ah suggestion for payload optimisation" commit broke that by exiting to state 7 before all the bytes are checksummed.

Reply:
Reverted [42fc8c04f](https://github.com/betaflight/betaflight/pull/12799/commits/42fc8c04fdbc436c9ef196b88f0764ffcbb9239f)